src/app/Plans/PlansPage.tsx

```diff
@@ -1,14 +1,125 @@
 import * as React from 'react';
-import { PageSection, Title } from '@patternfly/react-core';
-
-const PlansPage: React.FunctionComponent = () => (
-  <>
-    <PageSection>
-      <Title headingLevel="h1" size="lg">
-        Plans Page Title
-      </Title>
-    </PageSection>
-  </>
-);
+import {
+  Card,
+  PageSection,
+  CardBody,
+  EmptyState,
+  EmptyStateIcon,
+  Title,
+  Button,
+  Bullseye,
+  Spinner,
+  Tooltip,
+} from '@patternfly/react-core';
+import { AddCircleOIcon } from '@patternfly/react-icons';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import PlansTable from './components/PlansTable';
+import { IAddPlanDisabledObjModel } from './types';
+
+// TODO replace these with real state e.g. from redux
+import { MOCK_PLANS } from './mocks/plans.mock';
+import { MOCK_PROVIDERS } from '@app/Providers/mocks/providers.mock';
+const IsFetchingInitialPlans = false; // Fetching for the first time, not polling
+const migplans = MOCK_PLANS;
+const providers = MOCK_PROVIDERS;
+
+const PlansPage: React.FunctionComponent = () => {
+  const VMWareList = providers.map((x) => x.spec.type === 'vsphere');
+  const CNVList = providers.map((x) => x.spec.type === 'cnv');
+
+  const [isWizardOpen, toggleWizard] = React.useReducer(() => !isWizardOpen, false);
+
+  const [addPlanDisabledObj, setAddPlanDisabledObj] = React.useState<IAddPlanDisabledObjModel>({
```

Comment:
I know this stuff was copied from mig-ui, but this is one of the `useEffect` calls that isn't necessary. The only place this state changes is immediately after it is initialized. So `addPlanDisabledObj` doesn't need to be state, it can just be a variable that is initialized on each render. So we could replace lines 32-58 here with this:

```ts
let addPlanDisabledObj: IAddPlanDisabledObjModel = {
  isAddPlanDisabled: false,
  disabledText: 'Create a migration plan to select VMs to migrate to OpenShift virtualization.',
};

if (vmwareList.length < 1) {
  addPlanDisabledObj = {
    isAddPlanDisabled: true,
    disabledText: 'At least 1 VMWare provider is required to create a plan.',
  }
}

if (cnvList.length < 1) {
  addPlanDisabledObj = {
    isAddPlanDisabled: true,
    disabledText: 'At least 1 OpenShift provider is required to create a plan.',
  }
}
```

Reply:
Note that I also changed the capitalization of `VMWare` to `VMware` (lowercase W), apparently we've been getting that wrong :)

Reply:
Great, I love this. I effectively had been wondering about the state of `addPlanDisableObj` and assumed there were cases where the backend would be changed behind the curtain and we would need this to be reflected in the displayed screen. Thanks for clarifying.

The VMware capitalization has been hunting everyone (I know it's not just me) for years. It has to stop!!! ;)